src/backend/distributed/shared_library_init.c

```diff
@@ -2907,7 +2914,8 @@ ApplicationNameAssignHook(const char *newval, void *extra)
 	 * as reasonably possible, which is also why we extract global pids in the
 	 * AuthHook already (extracting doesn't require catalog access).
 	 */
-	if (FinishedStartupCitusBackend)
+	if (FinishedStartupCitusBackend &&
```

Comment:
I feel like now we can entirely remove `FinishedStartupCitusBackend` variable and the first part of this `&&` because when `IsTransactionState() || CachedLocalNodeIdIsValid())` is true, then we already cannot be in a state where StartupCitusBackend() has not been called yet?

Reply:
I think you had a valid point. Why not to remove FinishedStartupCitusBackend?